larray_editor/arrayadapter.py

```diff
@@ -101,34 +91,37 @@ def get_bg_value_2D(self, shape_2D):
     # XXX: or create two methods?:
     # - set_data (which reset the current filter)
     # - update_data (which sets new data but keeps current filter unchanged)
-    def set_data(self, data, bg_value=None, current_filter=None):
-        if data is None:
-            data = la.LArray([])
```

Comment:
why moving these 2 lines (if data is None: ...) in `ArrayEditorWidget`?

Reply:
Like I tried to explain in the commit comments: just because it makes the internal api simpler: you know what you get. In general, external API should accept as many useful things as possible, internal API should be as simple as possible.